mobile/library/common/jni/jni_interface.cc

```diff
@@ -1109,6 +1112,143 @@ Java_io_envoyproxy_envoymobile_engine_JniLibrary_registerStringAccessor(JNIEnv*
   return result;
 }
 
+void setString(JNIEnv* env, jstring java_string, EngineBuilder* builder,
+               EngineBuilder& (EngineBuilder::*setter)(std::string)) {
+  if (java_string) {
+    const char* native_java_string = env->GetStringUTFChars(java_string, nullptr);
+    std::string java_string_str = std::string(native_java_string);
+    if (!java_string_str.empty()) {
+      (builder->*setter)(java_string_str);
+      env->ReleaseStringUTFChars(java_string, native_java_string);
+    }
+  }
+}
+
+std::string to_string(JNIEnv* env, jbyteArray j_data) {
```

Comment:
nit: we have `setString()` and `to_string()` but we should probably have a single convention.